packages/pwa-kit-runtime/src/utils/ssr-server/configure-proxy.js

```diff
@@ -248,10 +249,13 @@ export const configureProxy = ({
             }
         },
 
-        // Rewrite the request's path to remove the /mobify/proxy/...
-        // prefix.
-        pathRewrite: {
-            [`^${proxyPath}`]: ''
+        // Rewrite the request's path to remove the /mobify/proxy/... prefix.
+        // This cannot be modified by any express middleware
+        // So we need to use the built in pathRewrite to remove the base path if present
+        pathRewrite: (path) => {
```

Comment:
hmm we are changing the signature of this method, no? before it is an object, now it is a function? how does this work before comparing to change in this PR?

Reply:
ie. An incoming request to the proxy may be:

`/mobify/proxy/api/product/shopper-products/v1/...`
The path rewrite removes the `/mobify/proxy/api` so that the outbound request to SCAPI is for `/product/shopper-products/v1/...`

Reply:
The reason we have to do this pathRewrite is because when http-proxy-middleware receives a request, it receives the original, unmodified path that includes the base path. It is not aware of any changes that Express has done to remove the base path.